dashboard/lib/services/user/upgrade_to_teacher.rb

```diff
@@ -0,0 +1,38 @@
+module Services
+  module User
+    class UpgradeToTeacher < Services::Base
+      attr_reader :user, :email, :email_preference
+
+      def initialize(user:, email:, email_preference: nil)
+        @user = user
+        @email = email
+        @new_attributes = email_preference || {}
+      end
+
+      def call
+        return true if user.teacher?
+        return false if email.blank?
+
+        user.user_type = ::User::TYPE_TEACHER
+        user.parent_email = nil
+
+        hashed_email = ::User.hash_email(email)
+
+        user.lti_roster_sync_enabled = true if ::Policies::Lti.lti?(user)
+
+        user.transaction do
+          if user.migrated?
+            user.update_primary_contact_info!(new_email: email, new_hashed_email: hashed_email)
+          else
+            @new_attributes[:email] = email
```

Comment:
Linked to the comment above, you can call `new_attributes` without the `@` once you've added it to the `attr_reader` list.

Reply:
I actually did have that question.